js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx

```diff
@@ -0,0 +1,333 @@
+
+type AssetCountsResult = {
+  countsByOwner: Record<string, number>;
+  countsByComputeKind: Record<string, number>;
+  countPerAssetGroup: CountPerGroupName[];
+  countPerCodeLocation: CountPerCodeLocation[];
+};
+
+type GroupMetadata = {
+  groupName: string;
+  repositoryLocationName: string;
+  repositoryName: string;
+};
+
+type CountPerGroupName = {
+  assetCount: number;
+  groupMetadata: GroupMetadata;
+};
+
+type CountPerCodeLocation = {
+  repoAddress: RepoAddress;
+  assetCount: number;
+};
+
+function buildAssetCountBySection(assets: AssetTableFragment[]): AssetCountsResult {
+  const assetCountByOwner: Record<string, number> = {};
+  const assetCountByComputeKind: Record<string, number> = {};
+  const assetCountByGroup: Record<string, number> = {};
+  const assetCountByCodeLocation: Record<string, number> = {};
+
+  assets
+    .filter((asset) => asset.definition)
+    .forEach((asset) => {
+      const assetDefinition = asset.definition!;
+      assetDefinition.owners.forEach((owner) => {
+        const ownerKey = owner.__typename === 'UserAssetOwner' ? owner.email : owner.team;
+        assetCountByOwner[ownerKey] = (assetCountByOwner[ownerKey] || 0) + 1;
+      });
+
+      const computeKind = assetDefinition.computeKind;
+      if (computeKind) {
+        assetCountByComputeKind[computeKind] = (assetCountByComputeKind[computeKind] || 0) + 1;
+      }
+
+      const groupName = assetDefinition.groupName;
+      const locationName = assetDefinition.repository.location.name;
+      const repositoryName = assetDefinition.repository.name;
+
+      if (groupName) {
+        const metadata: GroupMetadata = {
+          groupName,
+          repositoryLocationName: locationName,
+          repositoryName,
+        };
+        const groupIdentifier = JSON.stringify(metadata);
+        assetCountByGroup[groupIdentifier] = (assetCountByGroup[groupIdentifier] || 0) + 1;
+      }
+
+      const stringifiedCodeLocation = buildRepoPathForHuman(repositoryName, locationName);
+      assetCountByCodeLocation[stringifiedCodeLocation] =
+        (assetCountByCodeLocation[stringifiedCodeLocation] || 0) + 1;
+    });
+
+  const countPerAssetGroup = Object.entries(assetCountByGroup).map(([groupIdentifier, count]) => ({
+    assetCount: count,
+    groupMetadata: JSON.parse(groupIdentifier),
+  }));
+
+  return {
+    countsByOwner: assetCountByOwner,
+    countsByComputeKind: assetCountByComputeKind,
+    countPerAssetGroup,
+    countPerCodeLocation: Object.entries(assetCountByCodeLocation).map(([key, count]) => {
+      return {repoAddress: repoAddressFromPath(key)!, assetCount: count};
+    }),
+  };
+}
+
+interface AssetOverviewCategoryProps {
+  children: React.ReactNode;
+  assetsCount: number;
+}
+
+function getGreeting(timezone: string) {
+  const hour = Number(
+    new Date().toLocaleTimeString('en-US', {
+      hour: '2-digit',
+      hourCycle: 'h23',
+      timeZone: timezone === 'Automatic' ? browserTimezone() : timezone,
+    }),
```

Comment:
No need to change anything here, but a couple notes:

- For future reference, these formatters can be expensive to set up. It's often a good idea to create a (memoized or static) single formatter object for reuse and call `format` on it.
- The approach of parsing out the number from the string looks good to me. Alternatively, if we chose not to care about the user's timezone setting, you could use the `getHours()` method on the Date object.

Reply:
Thanks for the heads up here.

Which item exactly is the formatter? Is it the date object?